File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div496-next-greater-element-i/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div496-next-greater-element-i.java
class Solution {
    
    public static int find(int ele,int[] arr){
        for(int i=0;i<arr.length;i++){
            if(ele==arr[i]){
                for(int j=i+1;j<arr.length;j++){
                    if(arr[j]>ele)return arr[j];
                }
                return -1;
            }
        }
        return -1;
    }
    
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        /*
        Map<Integer,Integer> map=new HashMap<>();
        int n=nums2.length;
        int m=nums1.length;

        for(int i=0;i<n;i++){
            map.put(nums2[i],i);
        }

        int[] ans=new int[m];
        Arrays.fill(ans,-1);

        int k=0;
        for(int i=0;i<m;i++){
            int num=nums1[i];
            int j=map.getOrDefault(num,-1);
            if(j<0 || j>=n)k++;
            else{
                int max=nums2[j];
                System.out.println(max);
                for(int l=j+1;l<n;l++){
                   if(max<nums2[l]){
                       max=nums2[l];break;
                   }
                }

                if(max==nums2[j])ans[k++]=-1;
                else ans[k++]=max;
            }
        }
        return ans;
        
        */
        
        //2nd approach
        for(int i=0;i<nums1.length;i++){
            nums1[i]=find(nums1[i],nums2);
        }
        return nums1;
        
    }

        
}
    

File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div496-next-greater-element-i/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-polygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div496-next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
                
        for(int i=0;i<nums1.size();i++){
            int maxi=nums1[i];
            int digit = nums1[i];
            auto it =find(nums2.begin(), nums2.end(),digit);
            // cout <<*it<<endl;
            it+=1;
            while(it !=nums2.end()){
                if(*it>maxi){
                    maxi=*it;
                    break;
                } 
                it++;
            }
            if(maxi==nums1[i]) nums1[i]=-1;
            else nums1[i]=maxi;
        }
        return nums1;
    }
};